common/src/components/Pill/Pill.tsx

```diff
@@ -1,19 +1,69 @@
 import React from "react";
 
-export const Pill: React.FC<{ content: string }> = ({ content }) => {
+export interface PillProps extends React.HTMLProps<HTMLSpanElement> {
+  /** The style type of the element. */
+  color: "primary" | "secondary";
+  /** The style mode of the element. */
+  mode: "solid" | "outline";
+  /** Determines whether the element should be block level and 100% width. */
+  block?: boolean;
+  role?: "cell" | undefined;
+}
+
+const colorMap: Record<
+  "primary" | "secondary",
+  Record<"solid" | "outline", Record<string, string>>
+> = {
+  primary: {
+    solid: {
+      "data-h2-border": "b(darkpurple, all, solid, s)",
+      "data-h2-bg-color": "b(lightpurple)",
+      "data-h2-font-color": "b(white)",
+    },
+    outline: {
+      "data-h2-border": "b(darkpurple, all, solid, s)",
+      "data-h2-bg-color": "b(lightpurple[.1])",
+      "data-h2-font-color": "b(darkpurple)",
+    },
+  },
+  secondary: {
+    solid: {
+      "data-h2-border": "b(darknavy, all, solid, s)",
+      "data-h2-bg-color": "b(lightnavy)",
+      "data-h2-font-color": "b(white)",
+    },
+    outline: {
+      "data-h2-border": "b(darknavy, all, solid, s)",
+      "data-h2-bg-color": "b(lightnavy[.1])",
+      "data-h2-font-color": "b(darknavy)",
+    },
+  },
+};
+
+export const Pill: React.FC<PillProps> = ({
+  children,
+  color,
+  mode,
+  role,
+  block = false,
+  ...rest
+}): React.ReactElement => {
   return (
     <span
-      data-h2-border="b(darkpurple, all, solid, s)"
-      data-h2-display="b(block)"
+      data-h2-padding="b(all, xs)"
+      {...(block
+        ? { "data-h2-display": "b(block)" }
+        : { "data-h2-display": "b(inline-block)" })}
       data-h2-radius="b(m)"
-      data-h2-bg-color="b(lightpurple[.1])"
-      data-h2-padding="b(all, s)"
-      data-h2-font-color="b(darkpurple)"
       data-h2-font-size="b(caption)"
-      data-h2-margin="b(top-bottom, s)"
-      role="cell"
+      data-h2-font-family="b(sans)"
+      {...colorMap[color][mode]}
+      data-h2-margin="b(all, xxs)"
+      role={role || "cell"}
```

Comment:
Yeah, I set it that way because I wasn't sure what the other possible values of role should be.  Do you have a list of values?  Or should I just leave it as string?

Reply:
Yeah, I'm not sure what else it should be either, but I would either remove it from the component props, or set it to string (with a comment explaining it defaults to "cell")

Reply:
Haha, fair enough.  Having this prop is one of the acceptance criteria in the issue so I don't want to remove it but I have switched it to type _string_.  I guess we can easily constrain it again later if we want.  Good call on needing a comment.

Reply:
@tristan-orourke I don't think we need the `role="cell"` at all. The pill is just a word with design around it to make it look like a pill. The cell role is defined as "A cell in a tabular container." and is meant to be nested within the role row. I think we should remove it, since it's not meant for this.

Reply:
In [the original issue](https://github.com/GCTC-NTGC/gc-digital-talent/issues/359) one of the criteria was to add a prop to control the role attribute.  I think I'm hearing a consensus to remove that prop entirely?  It would easy enough to add back in later if we come up with a reason to use it.

Reply:
Yeah, seems like Yoni's right, and this isn't an appropriate use of `role="cell"`. Lets remove it from the pill component. If we wanted to use it somewhere specific, it could always be passed in along with the `rest` attributes.

Reply:
👍 Role prop removed.